### DISCUSSION TITLE
Optical Center Windows

### DISCUSSION THREAD

**Discussion by Down10** [rxhanson/Rectangle#1445] [Ideas]
Like most of users here, I love to have my windows arranged and aligned to my compulsive liking, and for the past decade, I have had the notion of aligning windows on my screen to the *optical center* of the screen, as opposed to the geometric center. What this means that smaller windows on a screen are set slightly higher than 50%, and this value is determined through the use of the golden ratio, rather than half. 

![Diagram depicting the optical center of a box, instead of the geometric center](https://github.com/user-attachments/assets/6838a358-453e-4449-b41e-cf516803e24f)

Apple themselves will align some dialogs and windows [higher than the absolute geometric center](https://developer.apple.com/documentation/appkit/nswindow/1419090-center) of the screen, so there is some precedence to this practice, which is rooted in the foundations of print and UI design.

I have already made a MacOS Shortcut to achieve this, which [anyone can download and explore for free here](https://www.icloud.com/shortcuts/85c23e57a0de4196ad2af5a0c23b460c). I myself use it alongside Rectangle.

Rectangle is terrific software and I think it would benefit with this unique function added. I have seen similar issues brought up in the past, but I'm not skilled enough as a programmer to add this in the app's code myself. Let me know what you think.

**Comment by Aaron-Rumpler** [comment]
Rectangle Pro does have an "Upper Center" command, but I've found that for bigger windows, it actually places the window lower than the "Center" command.

**Comment by rxhanson** [comment]
This can be configured in Rectangle with the centerProminently terminal command:

https://github.com/rxhanson/Rectangle/blob/main/TerminalCommands.md#add-an-extra-centering-with-prominence-command

(restart the app after executing a terminal command)

**Reply by Down10** [reply]
Interesting. I don't know what the float values mean, but I'll try them. Thanks!

**Reply by Aaron-Rumpler** [reply]
From the top of that page:
> A number of the commands below are for configuring keyboard shortcuts, and you'll need to know the key code and modifier flags integer values. You can use the free [key codes app](https://apps.apple.com/us/app/key-codes/id414568915) to help with this.

So you'd use that to get the key code and modifier for the keyboard shortcut you want, and substitute those values into the terminal command.

**Reply by Aaron-Rumpler** [reply]
@rxhanson Is `centerProminently` the same as "Upper Center" in Rectangle Pro?

**Reply by rxhanson** [reply]
Those are two different implementations, and looks like I have a bug in Upper Center. Thanks! I'll put in a fix that'll go out in the next release.
